cli/src/pcluster/validators/secret_validators.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+import base64
+import binascii
+
+from pcluster.aws.aws_api import AWSApi
+from pcluster.aws.common import AWSClientError
+from pcluster.validators.common import FailureLevel, Validator, _get_service_and_resource, _handle_arn_aws_client_error
+
+
+class ArnServiceAndResourceValidator(Validator):
+    """Validate that Arn is a valid ARN in given region."""
+
+    def _validate(self, arn: str, region: str, expected_service: str, expected_resource: str):
```

Comment:
Please put a TODO to possibly extend to dictionaries of pairs `{"allowed_service" : "allowed_resource"}`.

Reply:
Done!